### ISSUE TITLE
Cached observer with cached result does not update `statusAtom`

### ISSUE THREAD

**Issue by V-Mann-Nick** [jotaijs/jotai-tanstack-query#34]
For a while now I had noticed that when using `atomsWithQuery` relying on the `statusAtom` sometimes inexplicably the result from the jotai atom did not match the one showed by the react-query dev tools.

Today I have found a pattern to it and have created a little [reproduction repository](https://codesandbox.io/s/jotai-query-bug-reproduction-n1kvcr) for this issue:

- A question is some breed of cat or dog
- The answer is a jotai `atomsWithQuery` status atom
- First we see Question `labrador` -> Answer: `dog`
- When we click on "Run edge case" we set the question to `persian` which correctly yields `cat`
- We sleep for a second and the we set it back to `labrador`
- Unexpectedly the answer still remains `cat` and the `statusAtom` has not updated

Basically this is what happens:

- A query observer is created when the atom is created in the `observerAtom` ([here](https://github.com/jotaijs/jotai-tanstack-query/blob/v0.7.0/src/common.ts#L58-L59))
- The `baseStatusAtom` subscribes to react-query ([here](https://github.com/jotaijs/jotai-tanstack-query/blob/v0.7.0/src/common.ts#L71-L83))
- Now the query key changes which reuses the same `QueryObserver` instance using `setOptions` on it with `{ listeners: false }` ([here](https://github.com/jotaijs/jotai-tanstack-query/blob/v0.7.0/src/common.ts#L55)).

=> Now the issue occurs if the query result for this new query key is not stale and cached. React query immediately has a result, but because of `{ listeners: false }` this result is never brought to jotai's attention and the result is outdated.

I'm not sure why this flag was set but I can see that it had to do with some [fix](https://github.com/jotaijs/jotai-tanstack-query/commit/934111062c403853b13d5d0471c780b6a041cdf1) I don't fully understand.

In my investigation the removal of `{ listeners: false }` fixed my issue and I could not see any side effects.

What do you think?

**Comment by dai-shi** [comment]
Sounds like a bug or a limitation in the current impl design.
We need someone to help on it.
I think it's better to re-design from the ground: https://github.com/jotaijs/jotai-tanstack-query/issues/32#issuecomment-1523619697

**Comment by V-Mann-Nick** [comment]
Well I was able to fix the issue by removing the `{ listeners: false }` flag on [this line](https://github.com/jotaijs/jotai-tanstack-query/blob/v0.7.0/src/common.ts#L55) and I could make a quick PR to do that. I'm just not sure about the implications. There must have been a reason to add that flag.

I read your comment in the other issue. As much as I would like to help with small things, I don't feel competent enough with tanstack-query to revise the API of `jotai-tanstack-query`. For me personally it covers all my use cases since we got access to the `statusAtom` and except for this bug I have had no problems with it.
